[JIT] Refactor file:line:col to be less ugly

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #22180 [JIT] Load original SourceRanges on import
* #22179 [JIT] Write out SourceRanges on save()
* #22178 [JIT] Serialization methods for SourceRange and Source
* **#22177 [JIT] Refactor file:line:col to be less ugly**

Differential Revision: [D15981424](https://our.internmc.facebook.com/intern/diff/D15981424)

Commit:
[JIT] Refactor file:line:col to be less ugly

Commit:
Update on "[JIT] Refactor file:line:col to be less ugly"

[JIT] Refactor file:line:col to be less ugly

gh-metadata: pytorch pytorch 22177 gh/jamesr66a/6/head

Commit:
Update on "[JIT] Refactor file:line:col to be less ugly"

[JIT] Refactor file:line:col to be less ugly

gh-metadata: pytorch pytorch 22177 gh/jamesr66a/6/head

Commit:
Update on "[JIT] Refactor file:line:col to be less ugly"

[JIT] Refactor file:line:col to be less ugly

gh-metadata: pytorch pytorch 22177 gh/jamesr66a/6/head

zdevito commented on torch/csrc/jit/source_range.h:
```diff
@@ -117,6 +117,20 @@ struct CAFFE2_API SourceRange {
     return ss.str();
   }
 
+  c10::optional<std::tuple<std::string, size_t, size_t>> file_line_col() const {
```
perf: This is creating a copy of the filename string for _every_ line of the output.

Commit:
Update on "[JIT] Refactor file:line:col to be less ugly"

[JIT] Refactor file:line:col to be less ugly

gh-metadata: pytorch pytorch 22177 gh/jamesr66a/6/head

Commit:
Update on "[JIT] Refactor file:line:col to be less ugly"

[JIT] Refactor file:line:col to be less ugly

gh-metadata: pytorch pytorch 22177 gh/jamesr66a/6/head

Commit:
Update on "[JIT] Refactor file:line:col to be less ugly"

[JIT] Refactor file:line:col to be less ugly

gh-metadata: pytorch pytorch 22177 gh/jamesr66a/6/head
